http/submitattestations.go

```diff
@@ -18,35 +18,92 @@ import (
 	"context"
 	"encoding/json"
 	"errors"
+	"strings"
 
+	client "github.com/attestantio/go-eth2-client"
 	"github.com/attestantio/go-eth2-client/api"
-	"github.com/attestantio/go-eth2-client/spec/phase0"
+	"github.com/attestantio/go-eth2-client/spec"
 )
 
-// SubmitAttestations submits attestations.
-func (s *Service) SubmitAttestations(ctx context.Context, attestations []*phase0.Attestation) error {
+// SubmitAttestations submits versioned attestations.
+func (s *Service) SubmitAttestations(ctx context.Context, opts *api.SubmitAttestationsOpts) error {
 	if err := s.assertIsSynced(ctx); err != nil {
 		return err
 	}
+	if opts == nil {
+		return client.ErrNoOptions
+	}
+	if len(opts.Attestations) == 0 {
+		return errors.Join(errors.New("no attestations supplied"), client.ErrInvalidOptions)
+	}
+	attestations := opts.Attestations
+	unversionedAttestations, err := createUnversionedAttestations(attestations)
+	if err != nil {
+		return err
+	}
 
-	specJSON, err := json.Marshal(attestations)
+	specJSON, err := json.Marshal(unversionedAttestations)
 	if err != nil {
 		return errors.Join(errors.New("failed to marshal JSON"), err)
 	}
 
-	endpoint := "/eth/v1/beacon/pool/attestations"
+	endpoint := "/eth/v2/beacon/pool/attestations"
 	query := ""
 
-	if _, err := s.post(ctx,
+	headers := make(map[string]string)
+	headers["Eth-Consensus-Version"] = strings.ToLower(attestations[0].Version.String())
+	if _, err = s.post(ctx,
 		endpoint,
 		query,
-		&api.CommonOpts{},
+		&opts.Common,
 		bytes.NewReader(specJSON),
 		ContentTypeJSON,
-		map[string]string{},
+		headers,
 	); err != nil {
-		return errors.Join(errors.New("failed to submit beacon attestations"), err)
+		return errors.Join(errors.New("failed to submit versioned beacon attestations"), err)
 	}
 
 	return nil
 }
+
+func createUnversionedAttestations(attestations []*spec.VersionedAttestation) ([]any, error) {
+	var version *spec.DataVersion
```

Comment:
We don't need to use a pointer here, the enum has a `DataVersionUnknown` which we can use as the undefined value rather than `nil` in the later check, and that then removes the dereferencing.